### ISSUE TITLE
Roadmap: documentation infrastructure

### ISSUE THREAD

**Issue by denisrosset** [replab/replab#113]
I've been floating a few ideas around:

1. Use Jupyter to create/publish tutorials instead of the Matlab publish function #32

Advantages: supports Octave, WYSIWYG, supports LaTeX, it's an industry standard, can export back to `.m` files.

We can do that without replacing the rest of the website infrastructure, i.e. keep Jekyll and integrate Jupyter with it (worst case: using HTML files as we do currently).

2. Use some flavor of doctests to test inline code samples. Add more code samples in comments.

This can be done using an existing library #90 , this could also be achieved by code preprocessing (i.e. extract all code blocks and generate MOxUnit files).

3. Generate API documentation using Sphinx. Preliminary steps have been done: write a style guide, document public functions/methods/classes.

Basically, make the API accessible as part the website, even if it looks different, etc..

4. Correct the bugs in the release process. #1 #41 at least. Plus, the version on `master` should be called with a `-SNAPSHOT` suffix to state that the release is a moving target.

5. Migrate from Jekyll to Sphinx for all website related matters.

This is the point with the least benefit and the most risk. I don't have experience with Sphinx + Github pages, we should at least find projects that employ Sphinx for their GIthub website.

**Comment by jdbancal** [comment]
Thanks for putting all these points together! What do you think is most important? Here are my impressions about that:

Unless point 1 is needed in order to do the rest, I wouldn't suggest to make it a priority: we already have a temporarily working solution and we should be able to update it later without much consequences.

Point 3 is highly significant to the user. It looks to me like the most valuable point on the list.

Point 4, making sure the release process is smooth, seems quite important for the project: if we can rely on it, it should allows us to keep a consistent track of the progress, and therefore improve visibility in the project. The related bugs are also listed as part of the project https://github.com/replab/replab/projects/1 , which seems reasonable.

Point 2 would benefit the project by keeping examples up to date. Following the discussion in https://github.com/replab/replab/pull/90 we might not be fully ready for it yet. We could consider it as a follow-up of point 3.

Point 5 is a big change that relies on other points being done first, and will not provide very significant improvement (as you also summarized). Maybe we should keep it on hold for now.

**Comment by denisrosset** [comment]
Agree with the order of priority.

My own order would be 4) first (should be fast), then 3), then 2). If time remains, 1).

After things stabilize a bit, consider 5).

**Comment by denisrosset** [comment]
Referencing https://github.com/sphinx-contrib/matlabdomain/pull/96 from here.

**Comment by denisrosset** [comment]
We are currently working on this. We're keeping in mind the idea of moving everything to Sphinx, but not at this moment.

See #151
